File: lib/models/branch.dart
class Branch {
  final String id;
  final String slug;
  final String name;
  final String address;
  final double deliveryFee;
  final String deliveryTime;
  final String qrUrl;
  final String city;
  final String phone;
  final String openingHours;

  Branch({
    required this.id,
    required this.slug,
    required this.name,
    required this.address,
    required this.deliveryFee,
    required this.deliveryTime,
    this.qrUrl = '',
    this.city = '',
    this.phone = '',
    this.openingHours = '',
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['_id'] ?? json['id'] ?? '',
      slug: json['slug'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      deliveryFee: (json['deliveryFee'] ?? 0).toDouble(),
      deliveryTime: json['deliveryTime'] ?? '',
      qrUrl: json['qrUrl'] ?? '',
      city: json['city'] ?? '',
      phone: json['phone'] ?? '',
      openingHours: json['openingHours'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    '_id': id,
    'slug': slug,
    'name': name,
    'address': address,
    'deliveryFee': deliveryFee,
    'deliveryTime': deliveryTime,
    'qrUrl': qrUrl,
    'city': city,
    'phone': phone,
    'openingHours': openingHours,
  };
}

File: backend/models/Branch.js
const mongoose = require('mongoose');

const branchSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },
      slug: {
        type: String,
        required: true,
        trim: true,
        unique: true,
        lowercase: true,
      },
    address: {
      type: String,
      required: true,
    },
    city: {
      type: String,
      default: '',
    },
    phone: {
      type: String,
      default: '',
    },
    openingHours: {
      type: String,
      default: '',
    },
    qrUrl: {
      type: String,
      default: '',
    },
    deliveryFee: {
      type: Number,
      required: true,
      min: 0,
    },
    deliveryTime: {
      type: String,
      required: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Branch', branchSchema, 'branches');
